src/media/js/ratings.js

```diff
@@ -23,35 +23,29 @@ define('ratings',
         );
     }
 
-    function flagReview($reviewEl) {
-        var $modal = $('.report-spam');
+    function flagReview($review) {
+        // TODO: remove when reviews modal PR is merged.
+        var requests = require('requests');
+        var urls = require('urls');
 
-        if (!$modal.length) {
-            z.page.append(
-                nunjucks.env.render('ratings/report.html')
-            );
-            $modal = $('.report-spam');
-        }
+        z.page.append(nunjucks.env.render('_includes/flag_review_modal.html'));
+        $modal = $('mkt-prompt[data-modal="flag-review"]');
 
         $modal.one('click', '.menu a', utils._pd(function(e) {
-            var $actionEl = $reviewEl.find('.actions .flag');
-            $('.cloak').trigger('dismiss');
-            // L10n: The report is an abuse report for reviews
-            $actionEl.text(gettext('Sending report...'));
-            require('requests').post(
-                require('settings').api_url + urls.api.sign($reviewEl.data('report-uri')),
-                {flag: $(e.target).attr('href').replace('#', '')}
-            ).done(function() {
-                notify({message: gettext('Review flagged')});
+            var $actionEl = $review.find('.actions .flag');
+            $modal[0].dismissModal();
```

Comment:
This will cause a JS error if `!$modal` we might want to `return` earlier or wrap in `if ($modal)...`